File: __aacontest/_newestcontest/4b copy.py
from typing import List

class Solution:
    def minimumDistance(self, points: List[List[int]]) -> int:
        n = len(points)
        max_distance = 0
        
        # Calculate Manhattan distance between two points
        def manhattan_distance(p1, p2):
            return abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])
        
        # Function to find maximum distance by removing one point
        def find_max_distance_without_point(i):
            max_dist = 0
            for j in range(n):
                if j != i:
                    max_dist = max(max_dist, manhattan_distance(points[i], points[j]))
            return max_dist
        
        # Find maximum distance after removing each point
        for i in range(n):
            max_distance = max(max_distance, find_max_distance_without_point(i))
        
        return max_distance

# Test cases
solution = Solution()
print(solution.minimumDistance([[3,10],[5,15],[10,2],[4,4]])) # Output: 12
print(solution.minimumDistance([[1,1],[1,1],[1,1]])) # Output: 0

File: __aacontest/_newestcontest/4b copy.rb
def minimum_distance(points)
  # Transpose the points
  transposed_points = points.transpose
  
  # Find the minimum and maximum values along x and y axes
  min_x = transposed_points[0].min
  max_x = transposed_points[0].max
  min_y = transposed_points[1].min
  max_y = transposed_points[1].max
  
  # Calculate the diagonal distances
  diagonal_distances = [
    (max_x - min_x) + (max_y - min_y), # diagonal from top-left to bottom-right
    (max_x - min_x) + (min_y - max_y)  # diagonal from bottom-left to top-right
  ]
  
  # Return the maximum diagonal distance
  diagonal_distances.max
end



# Test cases
points1 = [[3,10],[5,15],[10,2],[4,4]]
points2 = [[1,1],[1,1],[1,1]]
points3 = [[3,2],[3,9],[7,10],[4,4],[8,10],[2,7]]

puts minimum_distance(points1) # Output: 13
puts minimum_distance(points2) # Output: 0
puts minimum_distance(points3) # Output: 8
